Mathlib/CategoryTheory/Monoidal/Action/Basic.lean

```diff
@@ -0,0 +1,354 @@
+* Canonical action of `Type u` on `u`-small cocomplete categories via the
+  copower.
+
+-/
+
+namespace CategoryTheory.MonoidalCategory
+
+variable (C D : Type*)
+
+variable [Category C] [Category D]
+/-- A class that carries the non-Prop data required to define a left action of a
+monoidal category `C` on a category `D`, to set up notations. -/
+class MonoidalLeftActionStruct [MonoidalCategoryStruct C] where
+  /-- The left action on objects. This is denoted `c ⊙ₗ d`. -/
+  actionObj : C → D → D
+  /-- The left action of a map `f : c ⟶ c'` in `C` on an object `d` in `D`.
+  If we are to consider the action as a functor `Α : C ⥤ D ⥤ D`,
+  this is (Α.map f).obj d`. This is denoted `f ⊵ₗ d` -/
+  actionHomLeft {c c' : C} (f : c ⟶ c') (d : D) :
+    actionObj c d ⟶ actionObj c' d
+  /-- The action of an object `c : C` on a map `f : d ⟶ d'` in `D`.
+  If we are to consider the action as a functor `Α : C ⥤ D ⥤ D`,
+  this is (Α.obj c).map f`. This is denoted `c ⊴ₗ f`. -/
+  actionHomRight (c : C) {d d' : D} (f : d ⟶ d') :
+    actionObj c d ⟶ actionObj c d'
+  /-- The action of a pair of maps `f : c ⟶ c'` and `d ⟶ d'`. By default,
+  this is defined in terms of `actionHomLeft` and `actionHomRight`. -/
+  actionHom {c c' : C} {d d' : D} (f : c ⟶ c') (g : d ⟶ d') :
+    actionObj c d ⟶ actionObj c' d' := actionHomLeft f d ≫ actionHomRight c' g
+  /-- The structural isomorphism `(c ⊗ c') ⊙ₗ d ≅ c ⊙ₗ (c' ⊙ₗ d)`. -/
+  actionAssocIso (c c' : C) (d : D) :
+    actionObj (c ⊗ c') d ≅ actionObj c (actionObj c' d)
+  /-- The structural isomorphism between `𝟙_ C ⊙ₗ d` and `d`. -/
+  actionUnitIso (d : D) : actionObj (𝟙_ C) d ≅ d
+
+namespace MonoidalLeftAction
+
+export MonoidalLeftActionStruct
+  (actionObj actionHomLeft actionHomRight actionHom actionAssocIso
+    actionUnitIso)
+
+-- infix priorities are aligned with the ones from `MonoidalCategoryStruct`.
+
+/-- Notation for `actionObj`, the action of `C` on `D`. -/
+scoped infixr:70 " ⊙ₗ " => MonoidalLeftActionStruct.actionObj
+
+/-- Notation for `actionHomLeft`, the action of `C` on morphisms in `D`. -/
+scoped infixr:81 " ⊵ₗ " => MonoidalLeftActionStruct.actionHomLeft
+
+/-- Notation for `actionHomRight`, the action of morphism in `C` on `D`. -/
+scoped infixr:81 " ⊴ₗ " => MonoidalLeftActionStruct.actionHomRight
+
+/-- Notation for `actionHom`, the bifunctorial action of morphisms in `C` and
+`D` on `- ⊙ₗ -`. -/
+scoped infixr:70 " ⊙ₗ " => MonoidalLeftActionStruct.actionHom
+
+/-- Notation for `actionAssocIso`, the structural isomorphism
+`- ⊗ - ⊙ₗ - ≅ - ⊙ₗ - ⊙ₗ -`. -/
+scoped notation "σ_ₗ " => MonoidalLeftActionStruct.actionAssocIso
+
+/-- Notation for `actionUnitIso`, the structural isomorphism `𝟙_ C ⊙ₗ - ≅ -`. -/
+scoped notation "υ_ₗ " => MonoidalLeftActionStruct.actionUnitIso
+/-- Notation for `actionUnitIso`, the structural isomorphism `𝟙_ C ⊙ₗ - ≅ -`,
+allowing one to specify the acting category. -/
+scoped notation "υ_ₗ["J"]" => MonoidalLeftActionStruct.actionUnitIso (C := J)
```

Comment:
Would not `αₗ` and `λₗ` be slightly better names as they parallel the notations in the case of the obvious action of a monoidal category on itself?